### ISSUE TITLE
Come up with a way of declaring multiple alternative implementations to a single check

### ISSUE THREAD

**Issue by felipesanches** [googlefonts/fontbakery#3436]
Follow-up to PR #3432 (I don't want to derail the conversation in there, nor block a PR merge with further feature ideas)

The PR mentioned above enables one to pass other file-types to FontBakery. The immediate consequence of that is that it would be possible to perform checking for the same kinds of problems but in different stages of a font project. For instance, there could be a GlyphsApp file source-level check for something that we already check on TTF binaries. In that case, the check-id, description and rationales would be the same, only the implementations would be specific to the file-type being checked. ("polymorphism" comes to mind)

I'm wondering how could our nice decorator-based syntax be expanded to allow such thing...

**Comment by graphicore** [comment]
Pattern matching on the type of the check arguments comes to my mind. Could be done by some kind of generic switching meta-check that wraps the actual implementations. Maybe a SKIP that is not reported/silent could help to remove noise when there's no implementation, though, it would be even nicer to have this figured out even before calling the check e.g. when calculating `profile.execution_order` (or have a filter to execution order before executing?).

**Comment by felipesanches** [comment]
I've merged the GlyphsApp checks even though their check ids are not definitive yet. They'll share the same id as the corresponding binary ttf checks once we support check polymorphism.

It seems that @simoncozens has been experimenting with this, so I will assign him to this issue. Feel free to reassign it back to me later if you need me to work further on this issue.

**Comment by felipesanches** [comment]
Milestone is set to FB v0.8.3 due to these temporary check-ids. If we postpone this to some later release, then we'll need to temporarily disable those experimental checks.

**Comment by simoncozens** [comment]
Yeah. I have this working. (See [https://github.com/simoncozens/fontbakery/tree/multiple-implementations-parking-lot](this branch).) The problem is that now checks are no longer a function, those code tests which don't use `CheckTester` don't work. This is why I was working on #3453.

**Comment by felipesanches** [comment]
thanks for doing that. The few code-tests that did not use `CheckTester` were left-overs I wanted to someday cleanup. Glad to see this finally happening now ;-)

**Comment by felipesanches** [comment]
PR #3453 (Use CheckTester everywhere) is now merged (via followup PR #3473), so we can resume on this. I will take a look at your branch and see if I can merge it myself. If I have a hard time doing so, I'll come back here to ask for help.
